### PR TITLE
Additional finishing processing when abandoning discussion

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Move some discussion finished logic to separate command
- Fix invalid test file name
- Gracefully handle errors when finishing abandoned discussions
- Minor refactoring
- Process finished when finishing abandoned discussion

### CODE REVIEW

File: app/commands/mentor/discussion/finish_abandoned.rb [hunk 1]
```diff
@@ -2,15 +2,26 @@ class Mentor::Discussion::FinishAbandoned
   include Mandate
 
   def call
+    discussions_to_abandon.each do |discussion|
+      finish!(discussion)
+    rescue StandardError => e
+      Bugsnag.notify(e)
```
Comment (ErikSchierboom) [inline app/commands/mentor/discussion/finish_abandoned.rb:8]: Just noted that we didn't yet gracefully handle errors

File: test/commands/mentor/discussion/finish_abandoned.rb
Comment (ErikSchierboom) [inline test/commands/mentor/discussion/finish_abandoned.rb:1]: This file was missing the `_test` suffix

File: app/commands/mentor/discussion/process_finished.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  def update_roles!
+    User::UpdateMentorRoles.defer(mentor)
+  end
+
+  def log_metric!
+    Metric::Queue.(:finish_mentoring, discussion.finished_at, discussion:, track:, user: student)
+  end
+
+  delegate :track, :student, :mentor, to: :discussion
```
Comment (iHiD) [inline app/commands/mentor/discussion/process_finished.rb]:
```suggestion
  delegate :track, :student, :mentor, :rating, to: :discussion
```

File: app/commands/mentor/discussion/process_finished.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+class Mentor::Discussion::ProcessFinished
+  include Mandate
+
+  initialize_with :discussion, :rating
```
Comment (iHiD) [inline app/commands/mentor/discussion/process_finished.rb]:
```suggestion
  initialize_with :discussion
```

**Pull request merged**